rasa/nlu/classifiers/sklearn_intent_classifier.py

```diff
@@ -225,40 +244,53 @@ def predict(self, X: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
         sorted_indices = np.fliplr(np.argsort(pred_result, axis=1))
         return sorted_indices, pred_result[:, sorted_indices]
 
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
+    def persist(self) -> None:
         """Persist this model into the passed directory."""
+        with self._model_storage.write_to(self._resource) as model_dir:
+            file_name = self.__class__.__name__
+            classifier_file_name = model_dir / f"{file_name}_classifier.pkl"
+            encoder_file_name = model_dir / f"{file_name}_encoder.pkl"
 
-        classifier_file_name = file_name + "_classifier.pkl"
-        encoder_file_name = file_name + "_encoder.pkl"
-        if self.clf and self.le:
-            io_utils.json_pickle(
-                os.path.join(model_dir, encoder_file_name), self.le.classes_
-            )
-            io_utils.json_pickle(
-                os.path.join(model_dir, classifier_file_name), self.clf.best_estimator_
-            )
-        return {"classifier": classifier_file_name, "encoder": encoder_file_name}
+            if self.clf and self.le:
+                io_utils.json_pickle(encoder_file_name, self.le.classes_)
+                io_utils.json_pickle(classifier_file_name, self.clf.best_estimator_)
 
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional[Metadata] = None,
-        cached_component: Optional["SklearnIntentClassifier"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "SklearnIntentClassifier":
+    ) -> SklearnIntentClassifierGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
         from sklearn.preprocessing import LabelEncoder
 
-        classifier_file = os.path.join(model_dir, meta["classifier"])
-        encoder_file = os.path.join(model_dir, meta["encoder"])
-
-        if os.path.exists(classifier_file):
-            classifier = io_utils.json_unpickle(classifier_file)
-            classes = io_utils.json_unpickle(encoder_file)
-            encoder = LabelEncoder()
-            encoder.classes_ = classes
-            return cls(meta, classifier, encoder)
-        else:
-            return cls(meta)
+        try:
+            with model_storage.read_from(resource) as model_dir:
+                file_name = cls.__name__
+                classifier_file = list(model_dir.glob(f"**/{file_name}_classifier.pkl"))
```

Comment:
why do we actually need to glob this at all? Don't we exactly know where to load it from as we explicitly hardcode the file names?

Reply:
Oh right - would it be enough to just replace this with `model_dir / f"{file_name}_classifier.pkl"` ?